### ISSUE TITLE
Problems printing in landscape

### ISSUE THREAD

**Issue by mark-nicholson** [twbs/bootstrap#25629]
Hi,
   I spent the afternoon trying to figure out why I cannot print bootstrap 4 pages in landscape in Chrome.

  I am running:
      - Ubuntu 16.04
      - Chrome:  64.0.3282.167 (Official Build) (64-bit)

My test file originates from the 'demo' baseline posted on your website. It actually uses NO bootstrap classes or functionality at all.   It is the simple inclusion of the public CSS file which introduces the issues.

  (Oddly, I tried to drag-and-drop the HTML file into this bug and it complains that HTML files are 'not supported' ...)

Sooooo, here it is:

<!DOCTYPE html>
<html lang="en">
  <head>
    
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    
    

    
    <link rel="stylesheet" href="/nih/bootstrap/4.0.0/bootstrap.css">

    <title>Basic</title>

  </head>
  
  <body>

    <div>
      <h3>This is a basic heading</h3>
      <p>Some generic test text</p>
    </div>

    <footer>
      <p id=date></p>
    </footer>
    
    <script type="text/javascript">
      var el = document.getElementById('date');
      var cdate = Date();
      el.innerHTML = cdate;
    </script>
    
    
    <script
       src="https://code.jquery.com/jquery-3.2.1.slim.min.js"
       integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN"
       crossorigin="anonymous"></script>

    
    <script
       src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"
       integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q"
       crossorigin="anonymous"></script>
     
    
    <script
       src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js"
       integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl"
       crossorigin="anonymous"></script>

  </body>
  
</html>

To reproduce:
   1. Load this html code on Chrome
   2. Hit Ctrl-P to go into "print" mode
   Watch carefully as the "Layout" option will very briefly appear, then be removed so you CANNOT select between "Portrait" and "Landscape" modes.

Now, I traced this pesky issue down to this patch:

--- bootstrap.css	2018-01-18 10:41:15.000000000 -0800
+++ nih/bootstrap/4.0.0/bootstrap.css	2018-02-14 15:48:45.268666092 -0800
@@ -8945,9 +8945,9 @@ a.text-dark:hover, a.text-dark:focus {
   h3 {
     page-break-after: avoid;
   }
-  @page {
+  /*@page {
     size: a3;
-  }
+  }*/
   body {
     min-width: 992px !important;
   }

Basically, there seems to be an errant @page {} definition in the basic CSS code.  There is NO reason for this as the framework has no way of predicting what page-size to expect.  (Not to mention A3 is a bit of an odd one to choose...)

For reasons I cannot explain, hard-coding my own @page {} block in my private CSS file would not "override" the one in bootstrap.css.

Hopefully this can be a quick patch to the main line.

  Firefox (58.0.2 64-bit) does not demonstrate the same issue.  It seems to be able to ignore the @page tag completely and resets the default to the printer's 'default page size' (in my case US-Letter)

**Comment by nguyencongvu** [comment]
I use the latest bootstrap and I have same issue, too. No options for layout!
Print another normal webpage (not bootstrap) is ok

**Comment by Morgul** [comment]
For those using the SCSS/SASS build (webpack, grunt, etc), you can just do this:

```scss
$print-page-size: auto;
```

Make sure to set it before you import anything from bootstrap.

**Comment by nip3o** [comment]
This issue also happens on the Bootstrap demo pages, for example on https://getbootstrap.com/docs/4.0/examples/album/. Note that it is possible to change the printing to landscape when printing using system dialog, but the result is completely broken.

**Comment by Kalten70** [comment]
I also had this problem with the latest Bootstrap v4.1.1. As a workaround, adding this css rule after Bootstrap js/css loaded has fixed it for me.

`@page { size: auto }`

This re-enables the layout option in Chrome.

- Mac OS X 10.13.4
- Chrome: 67.0.3396.87 (Official Build) (64-bit)

**Comment by eliezedeck** [comment]
This problem is also happening on Chrome for Windows: Version 67.0.3396.99 (Official Build) (64-bit)

- Using the _Save as PDF printer_: **no options whatsoever**
- Using the _Microsoft PDF printer_: there are options, but **no way to select Portrait/Landscape mode**

**Comment by MartijnCuppens** [comment]
We've removed the print hacks in `v5`. As for `v4`, I would indeed suggest using `@page { size: auto }`. Not planning on fixing this, since it might reintroduce other issues we had before.
